src/Cms/LicenseStatus.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace Kirby\Cms;
+
+use Kirby\Toolkit\I18n;
+
+/**
+ * @package   Kirby Cms
+ * @author    Nico Hoffmann <nico@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ *
+ * @codeCoverageIgnore
+ */
+enum LicenseStatus: string
+{
+	/**
+	 * The license is valid and active
+	 */
+	case Active = 'active';
+
+	/**
+	 * Only used for the demo instance
+	 */
+	case Demo = 'demo';
+
+	/**
+	 * The free feature period of
+	 * the license is over.
+	 */
+	case Inactive = 'inactive';
+
+	/**
+	 * The installation has an old
+	 * license (v1, v2, v3)
+	 */
+	case Legacy = 'legacy';
+
+	/**
+	 * The installation has no license or
+	 * the license cannot be validated
+	 */
+	case Missing = 'missing';
+
+	/**
+	 * Returns the dialog according to the status
+	 */
+	public function dialog(): string
+	{
+		return match ($this) {
+			static::Missing => 'registration',
+			default         => 'license'
+		};
+	}
+
+	/**
+	 * Returns the icon according to the status.
+     * The icon is used for the system view and
+     * in the license dialog.
+	 */
+	public function icon(): string
+	{
+		return match ($this) {
+			static::Missing  => 'key',
+			static::Legacy   => 'alert',
+			static::Inactive => 'clock',
+			static::Active   => 'check',
+			static::Demo     => 'preview',
+		};
+	}
+
+	/**
+	 * The info text is shown in the license dialog
+     * in the status row.
+	 */
+	public function info(string|null $end = null): string
+	{
+		return match ($this) {
+			static::Demo => 'This is a demo installation',
```

Comment:
Couldn't we also use translation keys for the demo mode strings? If a user of our demo changes the user language, this will then update as well.